src/parameterizations/stochastic/MOM_stochastics.F90

```diff
@@ -163,6 +165,14 @@ subroutine stochastics_init(dt, grid, GV, CS, param_file, diag, Time)
     nyT = grid%jed - grid%jsd + 1
     nxB = grid%iedB - grid%isdB + 1
     nyB = grid%jedB - grid%jsdB + 1
+    if(present(input_restart_file)) then
+       k = len_trim(input_restart_file)
+       i = index(input_restart_file, '.r.')
+       if (i>0) then
+          stoch_restfile = input_restart_file(1:i)//'r_stoch'//input_restart_file(i+2:k)
```

Comment:
Here i am setting the namelist variable.  I think that it would be better to just make this an additional parameter passed in to init_stochastic_physics_ocn